torch_geometric/nn/models/huggingface_hub_wrapper.py

```diff
@@ -0,0 +1,124 @@
+import json
+import logging
+import os
+from pathlib import Path
+from typing import Dict, Optional
+
+import requests
+from huggingface_hub import (
+    ModelCard,
+    ModelCardData,
+    PyTorchModelHubMixin,
+    hf_hub_download,
+)
+
+CONFIG_NAME = 'config.json'
+MODEL_HUB_ORGANIZATION = "pytorch_geometric"
+
+
+def hfhub_model(PyGModel, dataset=None):
```

Comment:
I originally thought we should treat this as a `Mixin` as well. For example, `SchNet(ModelHubMixin)` now provides a function `from_pretrained(..., dataset='QM9')`.